tests/Integration/PrestaShopBundle/Translation/Provider/TranslationsCatalogueProviderTest.php

```diff
@@ -0,0 +1,324 @@
+        $this->assertArrayHasKey('Pay by check', $messages['ModulesCheckpaymentShop']);
+        $this->assertSame(
+            [
+                'default' => 'Pay by check',
+                'xliff' => 'Payer par chèque',
+                'database' => null,
+            ],
+            $messages['ModulesCheckpaymentShop']['Pay by check']
+        );
+    }
+
+    /**
+     * The provider should retrieve all translations from files that
+     * look like `AdminSomething` or `ModulesSomethingAdmin`
+     */
+    public function testGetDomainCatalogueForModule()
+    {
+        $providerFactory = $this->createMock(ProviderFactory::class);
+
+        $providerFactory
+            ->expects($this->any())
+            ->method('build')
+            ->willReturn(
+                new ModulesProvider(
+                    $this->databaseReader,
+                    $this->getBuiltInModuleDirectory(),
+                    $this->getDefaultTranslationsDirectory(),
+                    $this->container->get('prestashop.translation.loader.legacy_file'),
+                    $this->container->get('prestashop.translation.extractor.legacy_module'),
+                    'checkpayment'
+                )
+            );
+
+        $provider = new TranslationsCatalogueProvider($providerFactory);
+        $messages = $provider->getDomainCatalogue(
+            new ModulesType('checkPaymentShop'),
+            'fr-FR',
+            'ModulesCheckpaymentShop'
+        );
+        $this->assertIsArray($messages);
+
+        // Check integrity of translations
+        $this->assertCount(19, $messages);
+        $this->assertArrayHasKey('default', $messages[0]);
+        $this->assertArrayHasKey('xliff', $messages[0]);
+        $this->assertArrayHasKey('database', $messages[0]);
+        $this->assertArrayHasKey('tree_domain', $messages[0]);
+        $this->assertSame([
+            'Modules',
+            'Checkpayment',
+            'Shop',
+        ], $messages[0]['tree_domain']);
+
+        // search a word
+        $messages = $provider->getDomainCatalogue(
+            new ModulesType('checkPaymentShop'),
+            'fr-FR',
+            'ModulesCheckpaymentShop',
+            ['payment']
+        );
+        $this->assertIsArray($messages);
+
+        // Check integrity of translations
+        $this->assertCount(3, $messages);
+        $this->assertArrayHasKey('default', $messages[0]);
+        $this->assertArrayHasKey('xliff', $messages[0]);
+        $this->assertArrayHasKey('database', $messages[0]);
+        $this->assertArrayHasKey('tree_domain', $messages[0]);
+        $this->assertSame([
+            'Modules',
+            'Checkpayment',
+            'Shop',
+        ], $messages[0]['tree_domain']);
+    }
+
+    protected function tearDown()
+    {
+        $langId = Language::getIdByIso('fr', true);
+        if ($langId) {
+            \Db::getInstance()->execute(
```

Comment:
This scares me
- if run together with another test using the same database, can break it (not isolated)
- if run on wrong database, can delete precious data (it happens)
- coupled to database structure and `DbCore` implementation so if one evolves this test might crash for wrong reasons

Reply:
@sowbiba